CMakeLists.txt

```diff
@@ -1,17 +1,50 @@
+cmake_minimum_required(VERSION 3.10)
+project(ChunkTypesProject CXX)
 
-# Min CMake version
-cmake_minimum_required(VERSION 3.20)
-
-set(SOURCE_DIR "source")
-set(INCLUDE_DIR "include")
-
-idf_component_register(
-	SRCS
-		${SOURCE_DIR}/BaseChunk.cpp
-		${SOURCE_DIR}/TimeChunk.cpp
-		${SOURCE_DIR}/UDPChunk.cpp
-		${SOURCE_DIR}/WAVChunk.cpp
-		${SOURCE_DIR}/WatchdogChunk.cpp
-	INCLUDE_DIRS
-		${INCLUDE_DIR}
-)
\ No newline at end of file
+# Set C++ standard
+set(CMAKE_CXX_STANDARD 11)
+
+# Main executable
+add_library(ChunkTypesLib
```

Comment:
You can define variables in cmake. As an example:

set(MY_VARIABLE "Hello, CMake!")

and then call it using:

${MY_VARIABLE}

I would suggest using varibales like SOURCE_DIR, SOURCE_DIR_TESTS
This should allow for quick changes to cmake if the repository structure ever changes

Reply:
An example of how this can be done is in the esp folder

Reply:
cool, have implemented some vars for dir names